consensus/ethash/consensus.go

```diff
@@ -310,6 +310,9 @@ func (ethash *Ethash) verifyHeader(chain consensus.ChainHeaderReader, header, pa
 	if diff := new(big.Int).Sub(header.Number, parent.Number); diff.Cmp(big.NewInt(1)) != 0 {
 		return consensus.ErrInvalidNumber
 	}
+	if chain.Config().IsShanghai(new(big.Int).SetUint64(header.Time)) {
```

Comment:
The same questions as clique.

Reply:
Maybe the check should be ethash doesn;t support Withdrawal EIP, but not the entire Shanghai fork

Reply:
same as above, makes no sense to have half forks